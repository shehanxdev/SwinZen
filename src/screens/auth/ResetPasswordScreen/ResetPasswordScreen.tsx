import { yupResolver } from '@hookform/resolvers/yup';
import React from 'react';
import { Controller, SubmitErrorHandler, SubmitHandler, useForm } from 'react-hook-form';
import { View } from 'react-native';

import { AccountLockIcon, Button, PasswordField, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextVariant } from '@sz/constants';
import { ResetPasswordFormValues } from '@sz/models';
import { resetPasswordValidationSchema } from '@sz/utils';

import { BaseAuthScreen } from '../components/BaseAuthScreen';

export function ResetPasswordScreen() {
  const {
    control,
    handleSubmit,
    setFocus,
    formState: { isSubmitted, errors },
  } = useForm<ResetPasswordFormValues>({ mode: 'onChange', resolver: yupResolver(resetPasswordValidationSchema) });

  const onResetPasswordFormInvalid: SubmitErrorHandler<ResetPasswordFormValues> = () => {
    console.log(errors);
    //TODO:: handle error
  };

  const onResetPasswordFormValid: SubmitHandler<ResetPasswordFormValues> = async formInput => {
    // TODO: API Integration
    try {
      console.log('formInput', formInput);
    } catch (error: any) {
      console.log('error', error);
    }
  };
  return (
    <BaseAuthScreen>
      <View style={tw`flex-1 justify-between`} testID="ResetPasswordScreenTestID">
        <View style={tw`mx-5 flex-1 justify-center content-center`}>
          <View style={tw`items-center`}>
            <View style={tw`mt-3 mb-5`}>
              <Text variant={TextVariant.SubTitle2SemiBold}>Reset password</Text>
            </View>
            <View style={tw`mb-13`}>
              <Text variant={TextVariant.Body2Regular}>
                Your new password must be different from your previously used passwords
              </Text>
            </View>
          </View>
          <Controller
            control={control}
            name="password"
            render={({ field: { value, onChange, onBlur, ref }, fieldState: { error, isTouched } }) => (
              <PasswordField
                ref={ref}
                label="Your New Password"
                leftIcon={<AccountLockIcon />}
                value={value}
                onChangeText={onChange}
                onBlur={onBlur}
                helperText={(isTouched || isSubmitted) && error?.message}
                helperTextColor={Color.Error.SzMain}
                error={(isTouched || isSubmitted) && error !== undefined}
                returnKeyType={'next'}
                onSubmitEditing={() => setFocus('confirmPassword')}
              />
            )}
          />
          <Controller
            control={control}
            name="confirmPassword"
            render={({ field: { value, onChange, onBlur, ref }, fieldState: { error, isTouched } }) => (
              <PasswordField
                ref={ref}
                label="Confirm Your New Password"
                leftIcon={<AccountLockIcon />}
                value={value}
                onChangeText={onChange}
                onBlur={onBlur}
                helperText={(isTouched || isSubmitted) && error?.message}
                helperTextColor={Color.Error.SzMain}
                error={(isTouched || isSubmitted) && error !== undefined}
                returnKeyType={'done'}
              />
            )}
          />
        </View>
        <View style={tw`items-center mb-5 mx-5`}>
          <View style={tw`mb-3`}>
            <Button
              onPress={handleSubmit(onResetPasswordFormValid, onResetPasswordFormInvalid)}
              title="Save new password"
            />
          </View>
        </View>
      </View>
    </BaseAuthScreen>
  );
}
