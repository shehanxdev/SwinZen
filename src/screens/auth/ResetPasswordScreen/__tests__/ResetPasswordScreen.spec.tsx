import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ResetPasswordScreen } from '../ResetPasswordScreen';

describe('ResetPasswordScreen Screen', () => {
  const testID = 'ResetPasswordScreenTestID';

  const getRenderedScreen = () => render(<ResetPasswordScreen />);
  it(`should render ResetPasswordScreen correctly`, () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the ResetPasswordScreen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
