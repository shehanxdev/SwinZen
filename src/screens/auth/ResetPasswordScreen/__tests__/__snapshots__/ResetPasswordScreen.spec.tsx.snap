// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordScreen Screen should render ResetPasswordScreen correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4279320343,
      4278650887,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "height": "100%",
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
              }
            }
            testID="ResetPasswordScreenTestID"
          >
            <View
              style={
                {
                  "alignContent": "center",
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "marginLeft": 20,
                  "marginRight": 20,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <View
                  style={
                    {
                      "marginBottom": 20,
                      "marginTop": 12,
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 28,
                        },
                        {
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontSize": 21,
                          "fontWeight": "600",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Reset password
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginBottom": 52,
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Your new password must be different from your previously used passwords
                  </Text>
                </View>
              </View>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#E9ECEF",
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "SourceSansPro-SemiBold",
                        "fontSize": 16,
                        "fontWeight": "600",
                        "textAlign": "auto",
                      },
                    ]
                  }
                >
                  Your New Password
                </Text>
                <View
                  style={
                    {
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                        },
                        {
                          "backgroundColor": "#283E36",
                          "borderColor": "#ffffff00",
                          "borderRadius": 10,
                          "borderWidth": 1,
                        },
                      ]
                    }
                    testID="text-input-outline"
                  />
                  <View>
                    <View
                      style={
                        [
                          {
                            "paddingBottom": 0,
                          },
                          {
                            "minHeight": 50,
                            "paddingTop": 8,
                          },
                        ]
                      }
                    >
                      <TextInput
                        editable={true}
                        maxFontSizeMultiplier={1.5}
                        multiline={false}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        onSubmitEditing={[Function]}
                        placeholderTextColor="#ADB5BD"
                        returnKeyType="next"
                        secureTextEntry={true}
                        selectionColor="#6C757D"
                        style={
                          [
                            {
                              "flexGrow": 1,
                              "margin": 0,
                              "paddingHorizontal": 14,
                              "zIndex": 1,
                            },
                            {
                              "height": 50,
                            },
                            {
                              "paddingBottom": 17,
                              "paddingTop": 17,
                            },
                            {
                              "color": "#ADB5BD",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontVariant": undefined,
                              "fontWeight": undefined,
                              "textAlign": "left",
                              "textAlignVertical": "center",
                            },
                            false,
                            {
                              "marginLeft": 44,
                              "marginRight": 44,
                              "paddingLeft": 0,
                              "paddingRight": 0,
                            },
                          ]
                        }
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "height": 24,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": 24,
                          },
                          {
                            "left": 12,
                            "top": 16,
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "disabled": false,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        hitSlop={
                          {
                            "bottom": 6,
                            "left": 6,
                            "right": 6,
                            "top": 6,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          [
                            {
                              "overflow": "hidden",
                            },
                            [
                              {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "margin": 6,
                                "overflow": "hidden",
                              },
                              {
                                "borderRadius": 18,
                                "height": 36,
                                "width": 36,
                              },
                              undefined,
                              {
                                "marginTop": "50%",
                              },
                            ],
                          ]
                        }
                      >
                        <View>
                          <SvgMock
                            fill="#ADB5BD"
                            height={20}
                            width={20}
                          />
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "height": 24,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": 24,
                          },
                          {
                            "right": 12,
                            "top": 16,
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "disabled": false,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        hitSlop={
                          {
                            "bottom": 6,
                            "left": 6,
                            "right": 6,
                            "top": 6,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          [
                            {
                              "overflow": "hidden",
                            },
                            [
                              {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "margin": 6,
                                "overflow": "hidden",
                              },
                              {
                                "borderRadius": 18,
                                "height": 36,
                                "width": 36,
                              },
                              undefined,
                              {
                                "marginTop": "50%",
                              },
                            ],
                          ]
                        }
                      >
                        <View>
                          <SvgMock
                            fill="#ADB5BD"
                            height={20}
                            width={20}
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    [
                      {
                        "color": "#E35F5F",
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "textAlign": "auto",
                      },
                    ]
                  }
                />
              </View>
              <View>
                <Text
                  style={
                    [
                      {
                        "color": "#E9ECEF",
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "SourceSansPro-SemiBold",
                        "fontSize": 16,
                        "fontWeight": "600",
                        "textAlign": "auto",
                      },
                    ]
                  }
                >
                  Confirm Your New Password
                </Text>
                <View
                  style={
                    {
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 0,
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 6,
                        },
                        {
                          "backgroundColor": "#283E36",
                          "borderColor": "#ffffff00",
                          "borderRadius": 10,
                          "borderWidth": 1,
                        },
                      ]
                    }
                    testID="text-input-outline"
                  />
                  <View>
                    <View
                      style={
                        [
                          {
                            "paddingBottom": 0,
                          },
                          {
                            "minHeight": 50,
                            "paddingTop": 8,
                          },
                        ]
                      }
                    >
                      <TextInput
                        editable={true}
                        maxFontSizeMultiplier={1.5}
                        multiline={false}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        placeholderTextColor="#ADB5BD"
                        returnKeyType="done"
                        secureTextEntry={true}
                        selectionColor="#6C757D"
                        style={
                          [
                            {
                              "flexGrow": 1,
                              "margin": 0,
                              "paddingHorizontal": 14,
                              "zIndex": 1,
                            },
                            {
                              "height": 50,
                            },
                            {
                              "paddingBottom": 17,
                              "paddingTop": 17,
                            },
                            {
                              "color": "#ADB5BD",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontVariant": undefined,
                              "fontWeight": undefined,
                              "textAlign": "left",
                              "textAlignVertical": "center",
                            },
                            false,
                            {
                              "marginLeft": 44,
                              "marginRight": 44,
                              "paddingLeft": 0,
                              "paddingRight": 0,
                            },
                          ]
                        }
                        underlineColorAndroid="transparent"
                      />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "height": 24,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": 24,
                          },
                          {
                            "left": 12,
                            "top": 16,
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "disabled": false,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        hitSlop={
                          {
                            "bottom": 6,
                            "left": 6,
                            "right": 6,
                            "top": 6,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          [
                            {
                              "overflow": "hidden",
                            },
                            [
                              {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "margin": 6,
                                "overflow": "hidden",
                              },
                              {
                                "borderRadius": 18,
                                "height": 36,
                                "width": 36,
                              },
                              undefined,
                              {
                                "marginTop": "50%",
                              },
                            ],
                          ]
                        }
                      >
                        <View>
                          <SvgMock
                            fill="#ADB5BD"
                            height={20}
                            width={20}
                          />
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "height": 24,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": 24,
                          },
                          {
                            "right": 12,
                            "top": 16,
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          {
                            "disabled": false,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        focusable={true}
                        hitSlop={
                          {
                            "bottom": 6,
                            "left": 6,
                            "right": 6,
                            "top": 6,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          [
                            {
                              "overflow": "hidden",
                            },
                            [
                              {
                                "alignItems": "center",
                                "justifyContent": "center",
                                "margin": 6,
                                "overflow": "hidden",
                              },
                              {
                                "borderRadius": 18,
                                "height": 36,
                                "width": 36,
                              },
                              undefined,
                              {
                                "marginTop": "50%",
                              },
                            ],
                          ]
                        }
                      >
                        <View>
                          <SvgMock
                            fill="#ADB5BD"
                            height={20}
                            width={20}
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <Text
                  style={
                    [
                      {
                        "color": "#E35F5F",
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "textAlign": "auto",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "marginBottom": 20,
                  "marginLeft": 20,
                  "marginRight": 20,
                }
              }
            >
              <View
                style={
                  {
                    "marginBottom": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      {
                        "backgroundColor": "#69AA67",
                        "borderColor": "#ffffff00",
                        "borderRadius": 10,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "elevation": 0,
                        "justifyContent": "center",
                        "minWidth": "auto",
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "disabled": false,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          {
                            "borderRadius": 10,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            {
                              "height": 48,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                              },
                              {
                                "textAlign": "left",
                              },
                              [
                                {
                                  "letterSpacing": 1,
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                {
                                  "marginHorizontal": 8,
                                },
                                false,
                                {
                                  "color": "#6200ee",
                                  "fontFamily": "System",
                                  "fontWeight": "500",
                                },
                                {
                                  "fontFamily": "System",
                                  "fontWeight": "500",
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#F8F9FA",
                                  "lineHeight": 24,
                                },
                                {
                                  "fontFamily": "SourceSansPro-SemiBold",
                                  "fontSize": 16,
                                  "fontWeight": "600",
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Save new password
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
